### PR TITLE
Security plugin support

### PR DESCRIPTION
**Pull request by florianvazelle**
### Description
Added support for the Security API.

### Issues Resolved
The issue of managing roles, users... from the python client.  
This issue is the subject of a feature request at https://github.com/opensearch-project/opensearch-py/issues/110.

### CODE REVIEW

**Review by dblock — COMMENTED** [review]
Looks good!

Can we add something to https://github.com/opensearch-project/opensearch-py/blob/main/USER_GUIDE.md#using-plugins as part of this PR?

**Comment by florianvazelle** [comment]
I pushed some new commits to fix my tests that never passed, because I hadn't quite understood the test suite, which isn't very clear. I added the tests in an async version. 

Yes, I can add something to the USER_GUIDE, but the security API has a lot of enpoints, if I add examples for users and roles, is that enough?

File: .ci/run-opensearch.sh [hunk 1]
```diff
@@ -81,7 +81,7 @@ END
     docker run \
       --name "$node_name" \
       --network "$network_name" \
-      --env "ES_JAVA_OPTS=-Xms1g -Xmx1g" \
+      --env "OPENSEARCH_JAVA_OPTS=-Xms1g -Xmx1g" \
```
Comment (dblock) [inline .ci/run-opensearch.sh:84]: Thanks for this.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Added index lifecycle guide ([#362](https://github.com/opensearch-project/opensearch-py/pull/362))
 - Added 'point in time' APIs to the pyi files in sync and async client ([#378](https://github.com/opensearch-project/opensearch-py/pull/378))
 - Added MacOS and Windows CI workflows ([#390](https://github.com/opensearch-project/opensearch-py/pull/390))
+- Added support for the Security API ([#399](https://github.com/opensearch-project/opensearch-py/pull/399))
```
Comment (dblock) [inline CHANGELOG.md:11]: Let's say "security plugin"

File: USER_GUIDE.md [hunk 3]
```diff
@@ -420,6 +425,66 @@ query = {
 response = client.plugins.alerting.acknowledge_alert(query)
 print(response)
 ```
+
+### Security plugin
+
+#### **Creating a role**
```
Comment (dblock) [inline USER_GUIDE.md:431]: I think another PR was getting rid of these `**` bolding, let's remove those everywhere?

**Review by dblock — CHANGES_REQUESTED** [review]
Looks great! Split out security plugin tests in its own file(s). Otherwise I think this is good enough.

**Review by dblock — APPROVED** [review]

**Comment by saimedhi** [comment]
Hello @florianvazelle, can you please move Security Client from plugins to clients. example: (opensearchpy/client/security.py) like in PR https://github.com/opensearch-project/opensearch-py/pull/439. All the security features are considered as part of core features of OS. So, kindly let me know if you are willing to make the following changes. This is crucial and had to be done before next release. Thanks for your contributions.

**Pull request merged**